canopen/node/local.py

```diff
@@ -44,15 +47,15 @@ def associate_network(self, network):
         network.subscribe(self.sdo.rx_cobid, self.sdo.on_request)
         network.subscribe(0, self.nmt.on_command)
 
-    def remove_network(self):
+    def remove_network(self) -> None:
         self.network.unsubscribe(self.sdo.rx_cobid, self.sdo.on_request)
         self.network.unsubscribe(0, self.nmt.on_command)
-        self.network = None
-        self.sdo.network = None
-        self.tpdo.network = None
-        self.rpdo.network = None
-        self.nmt.network = None
-        self.emcy.network = None
+        self.network = canopen.network._DummyNetwork()
+        self.sdo.network = self.network
+        self.tpdo.network = self.network
+        self.rpdo.network = self.network
+        self.nmt.network = self.network
+        self.emcy.network = self.network
```

Comment:
Consider to create a singleton placeholder in `canopen/network.py`, import the singleton here, instead of creating new objects all the time.

Ditto for `canopen/remote.py`, `canopen/pdo/base.py`, and `canopen/sdo/base.py`.

Then you can do something like this:

```suggestion
        self.network = _NetworkPlaceholder
        self.sdo.network = _NetworkPlaceholder
        self.tpdo.network = _NetworkPlaceholder
        self.rpdo.network = _NetworkPlaceholder
        self.nmt.network = _NetworkPlaceholder
        self.emcy.network = _NetworkPlaceholder
```

Reply:
What to name the singleton then?  I'm leaning towards: `_EmptyNetwork = _UninitializedNetwork()`.

Reply:
How about one of these?

- `_UNINITIALIZED_NETWORK`
- `_EMPTY_NETWORK`
- `_NETWORK_PLACEHOLDER`